Node.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BST
+{
+    class Node
+    {
+        public int key;
+        public Node left, right;
+
+        public Node(int item)
+        {
+            key = item;
+            left = right = null;
+        }
+        public override string ToString()
+        {
+            return "element" + " " + right.key + " " + left.key + " " + left.left.key;

```

Comment:
What does this mean? This could potentially lead to a Null pointer exception, if 'left' class attribute is not initialized.
If you want to print out this node (leaf), just go for:
`return "element " + this + " " + left + " " + right ;`
@domanttas @JulitaV I'm not too familiar with C#, would ^^^ count?

Reply:
It would certainly lead to NullPointerException in some cases. I do not know what would be spitted out to screen by printing 'this' in C#, but yeah, it is a much safer solution. However if intention was to print out only keys, maybe just the last part of original method could be removed? Either way, I'm not too sure what is the purpose of having method which prints nodes in Node class, shouldn't we be able to see whole tree printed out?

Reply:
BTW, this method is never used (or I'm blind).